esp/services/ws_fileio/ws_fileioservice.cpp

```diff
@@ -34,128 +35,56 @@ void CWsFileIOEx::init(IPropertyTree *cfg, const char *process, const char *serv
     tpWrapper.getTpDropZones(9999, nullptr, false, allTpDropZones); //version 9999: get the latest information about dropzone
 }
 
-bool CWsFileIOEx::CheckServerAccess(const char* targetDZNameOrAddress, const char* netAddrReq, const char* relPath, StringBuffer& netAddr, StringBuffer& absPath)
+void CWsFileIOEx::validateDropZoneAccess(IEspContext &context, const char *targetDZNameOrHost, const char *hostReq, SecAccessFlags permissionReq,
+    const char *fileNameWithRelPath, CDfsLogicalFileName &dlfn)
 {
-    if (!targetDZNameOrAddress || (targetDZNameOrAddress[0] == 0) || !relPath || (relPath[0] == 0))
-        return false;
+    if (containsRelPaths(fileNameWithRelPath)) //Detect a path like: a/../../../f
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file path %s", fileNameWithRelPath);
 
-    if (containsRelPaths(relPath)) //Detect a path like: a/../../../f
-        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file path %s", relPath);
-
-#ifdef _CONTAINERIZED
-    bool isIp4Req = isIPAddress(netAddrReq);
-    ForEachItemIn(i, allTpDropZones)
-    {
-        IConstTpDropZone& dropZone = allTpDropZones.item(i);
-        if (!dropZone.getECLWatchVisible())
-            continue;
-
-        const char* name = dropZone.getName();
-        if (isEmptyString(name) || !streq(targetDZNameOrAddress, name))
-            continue;
-
-        const char* prefix = dropZone.getPath();
-        if (isEmptyString(prefix))
-            continue;
-
-        IArrayOf<IConstTpMachine>& tpMachines = dropZone.getTpMachines();
-        ForEachItemIn(ii, tpMachines)
-        {
-            IConstTpMachine& tpMachine = tpMachines.item(ii);
-            if (!isEmptyString(netAddrReq) && !matchNetAddressRequest(netAddrReq, isIp4Req, tpMachine))
-                continue;
-
-            netAddr.set(tpMachine.getNetaddress());
-            absPath.set(prefix);
-            addPathSepChar(absPath);
-            absPath.append(relPath);
-            return true;
-        }
-    }
-#else
-    netAddr.clear();
-    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
-    Owned<IConstEnvironment> env = factory->openEnvironment();
-    Owned<IConstDropZoneInfo> dropZoneInfo = env->getDropZone(targetDZNameOrAddress);
-    if (!dropZoneInfo || !dropZoneInfo->isECLWatchVisible())
+    Owned<IPropertyTree> dropZone = getDropZonePlane(targetDZNameOrHost);
+    if (!dropZone) //The targetDZNameOrHost could be a dropzone host.
     {
-        if (stricmp(targetDZNameOrAddress, "localhost")==0)
-            targetDZNameOrAddress = ".";
-
-        Owned<IConstDropZoneInfoIterator> dropZoneItr = env->getDropZoneIteratorByAddress(targetDZNameOrAddress);
-        ForEach(*dropZoneItr)
+        Owned<IPropertyTreeIterator> planes = getDropZonePlanesIterator(nullptr);
+        ForEach(*planes)
         {
```

Comment:
why isn't this using findDropZonePlane ?

Reply:
The path used for findDropZonePlane() is an absolute path. The path here is a relative path (from the DestRelativePath).

Reply:
ok, but it is very similar code..
It would be better if findDropZonePlane was reused, and this code simplified.
I suggest changing findDropZonePlane, so that it allows path to be null, and returns a host only match.

Also, isIPAddress(targetDZNameOrHost) is currently called in the loop, the above change would avoid that,
but more importantly, I'm not sure you should set ipMatch=false if 'targetDZNameOrHost' is an IP..
That would mean that if a host in a dropzone resolved to the IP - it would not match it.